### PR TITLE
Add nasa images to database: Alex

### LINKED ISSUE #189: Integrate the NASA image of the day and ISS API calls into the dashboard script.
As a developer working on the StarWatch project,
I want to integrate the NASA Image of the Day API and the ISS location API into the dashboard,
so that users can view daily space imagery and real-time International Space Station (ISS) data alongside the existing astronomical information.

### PR DESCRIPTION
**Pull request by AlexOBarnes**
This closes #189 

I have:
Written tests for the nasa pipeline 
Implemented the pipeline to pass those tests

### CODE REVIEW

File: dashboard/test_image_pipeline.py
Comment (olliejth) [inline dashboard/test_image_pipeline.py:1]: Nice test classes.

File: dashboard/image_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+'''Checks for todays NASA image of day and the current ISS coordinates'''
+import sys
+import os
+sys.path.append(os.path.abspath(os.path.join(
+    os.path.dirname(__file__), '..', 'weekly-openmeteo')))
+from os import environ as ENV
+import logging
+from datetime import datetime as dt
```
Comment (olliejth) [inline dashboard/image_pipeline.py:8]: Good idea to import it like this.

File: dashboard/image_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        else:
+            raise ValueError(f'Could not obtain {KEYS[clean_data.index(None)]}')
+    raise APIError('Unsuccessful request.', response.status_code)
+
+
+def load_image(data: list) -> None:
+    '''Loads the nasa image onto the database'''
+    query = '''INSERT INTO image (image_date,image_name,image_url)
+    VALUES (%s,%s,%s)'''
+    with get_connection() as conn:
```
Comment (olliejth) [inline dashboard/image_pipeline.py:56]: Efficient use of with.

**Review by olliejth — APPROVED** [review]
Nice code, consistent styling and extensive test suite. Some missing functions that will need to be filled in later but good idea to break this up into smaller PRs.

File: dashboard/image_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from dotenv import load_dotenv
+from requests import get
+from psycopg2 import connect
+from api_error import APIError
+from extract import get_connection
+
+KEYS = ['data','title','url']
+
+
+
```
Comment (CalebMathers) [inline dashboard/image_pipeline.py:18]: Lots of whitespace here could be reduced a little

File: dashboard/image_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    ...
+
+def load_iss_data() -> None:
+    '''Uploads the data into the database'''
+    ...
+
+
+def iss_pipeline() -> None:
+    '''Runs the ISS pipeline to obtain the current ISS location'''
+    ...
```
Comment (CalebMathers) [inline dashboard/image_pipeline.py:86]: Good to see you're writing function declarations and planning out the script

File: dashboard/image_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    VALUES (%s,%s,%s)'''
+    with get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute(query,data)
+            conn.commit()
+
+def nasa_pipeline() -> None:
+    '''Runs the nasa pipeline to upload the nasa image for the day.'''
+    if not has_nasa_image():
+        logging.info('No nasa image found for today.')
```
Comment (CalebMathers) [inline dashboard/image_pipeline.py:64]: Good, informative, logging

File: dashboard/test_image_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    @patch('image_pipeline.get_nasa_image')
+    @patch('image_pipeline.has_nasa_image')
+    def test_nasa_pipeline_image_already_exists(self, mock_has_nasa_image, mock_get_nasa_image, mock_load_image):
+        '''Tests that a true response does not call the ETL pipeline'''
+        mock_has_nasa_image.return_value = True
+
+        nasa_pipeline()
+
+        mock_get_nasa_image.assert_not_called()
+        mock_load_image.assert_not_called()
```
Comment (CalebMathers) [inline dashboard/test_image_pipeline.py:134]: Good to make sure that certain things won't run when they shouldn't

File: dashboard/image_pipeline.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    WHERE image_name ILIKE '%nasa_%' AND
+     image_date = CURRENT_DATE """
+    with get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute(query)
+            data = cur.fetchone()
+    return bool(data[0])
+
+
+def extract_time(time_str: str) -> dt:
```
Comment (CalebMathers) [inline dashboard/image_pipeline.py:31]: Great job splitting functions up so they are small and testable

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**